dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+
+import com.liferay.portal.model.User;
+import com.liferay.util.SystemProperties;
+
+
+public class VelocityUtil {
+
+	private static VelocityEngine ve = null;
+	private static boolean DEFAULT_PAGE_TO_DEFAULT_LANGUAGE = LanguageWebAPI.canDefaultPageToDefaultLanguage();
+	
+	private synchronized static void init(){
+		if(ve != null)
+			return;
+		ve = new VelocityEngine();
+		try{
+			ve.init(SystemProperties.getProperties());
+
+			Logger.debug(VelocityUtil.class, SystemProperties.getProperties().toString());
+		}catch (Exception e) {
+			Logger.error(VelocityUtil.class,e.getMessage(),e);
+		}
+	}
+	
+	public static VelocityEngine getEngine(){
+		if(ve == null){
+			init();
+			if(ve == null){
+				Logger.fatal(VelocityUtil.class,"Velocity Engine unable to initialize : THIS SHOULD NEVER HAPPEN");
+				throw new DotRuntimeException("Velocity Engine unable to initialize : THIS SHOULD NEVER HAPPEN");
+			}
+		}
+		return ve;
+	}
+	/**
+	 * Changes $ and # to velocity escapes.  This is helps filter out velocity code injections.
+	 * @param s 
+	 * @return
+	 */
+	public static String cleanVelocity(String s) {
+		if (s==null) {
+			return null;
+		}
+		s=s.replace("$", "${esc.dollar}");
+		s=s.replace("#", "${esc.hash}");
+		return s;
+	}
+
+
+	
+	public String parseVelocity(String velocityCode, Context ctx){
+		VelocityEngine ve = VelocityUtil.getEngine();
+		StringWriter stringWriter = new StringWriter();
+		try {
+		   ve.evaluate(ctx, stringWriter, "VelocityUtil:parseVelocity", velocityCode);
+		}catch (Exception e) {
+		Logger.error(this,e.getMessage(),e);
+		}
+		return stringWriter.toString(); 
+		
+	}
+
+
+	public static String convertToVelocityVariable(final String variable, boolean firstLetterUppercase){
+		
+
+	      return (firstLetterUppercase) 
+	              ? StringUtils.camelCaseUpper(variable)
+	              : StringUtils.camelCaseLower(variable);
+		
+	}
+	
+	
+	public static Boolean isNotAllowedVelocityVariableName(String variable){
+		
+
+		String [] notallwdvelvars={"inode","type", "modDate", "owner", "ownerCanRead", "ownerCanWrite", "ownerCanPublish",
+				"modUser", "working", "live", "deleted", "locked","structureInode", "languageId", "permissions",
+				"identifier", "conHost", "conFolder", "Host", "folder"}; 
+		Boolean found=false;
+		for(String notallowed : notallwdvelvars){
+			 if(variable.equalsIgnoreCase(notallowed)){
+				 found=true;
+			 }
+			
+		}
+		return found;
+	}
+
+	/**
+	 * Returns a basic Velocity Context without any toolbox or request
+	 * response, session objects;
+	 * @return
+	 */
+
+	public static Context getBasicContext() {
+		Context context = new VelocityContext();
+		context.put("UtilMethods", new UtilMethods());
+		context.put("PortletURLUtil", new PortletURLUtil());
+		context.put("quote", "\"");
+		context.put("pounds", "##");
+		context.put("return", "\n");
+		context.put("velocityContext", context);
+		context.put("language", "1");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "language" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)